File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    return firstSolution(array, commands);
}

const firstSolution = (array, commands) => commands
    .map(([startIndex, endIndex, targetIndex]) => {
        return array.slice(startIndex - 1, endIndex).sort((x, y) => x - y)[targetIndex - 1]
    });

File: 프로그래머스/1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = firstSolution(array, commands)
        return answer
    }
    
    private fun firstSolution(
        array: IntArray,
        commands: Array<IntArray>
    ): IntArray {
        val answer = mutableListOf<Int>()
        for(command in commands) {
            // 1. (command[0] - 1) ~ command[1] sub => 문제에서는 1번째가 시작점이므로 탐색 시 command[0] - 1로 계산
            val subArray = array.copyOfRange(command[0] - 1, command[1])
            // 2. 오름 차순 정렬
            subArray.sort()
            // 3. K번째 요소 저장
            answer += subArray[command[2] - 1]
        }
        return answer.toIntArray()
    }
}